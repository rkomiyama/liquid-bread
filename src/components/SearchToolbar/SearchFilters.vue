<template>
  <v-card>
    <v-subheader>Search filters</v-subheader>
    <v-card-text class="pt-0">
      <v-layout wrap justify-space-between>
        <v-flex lg3 md3 sm5>
          <v-text-field v-model="filters.abvLt" label="ABV (less than)" hide-details></v-text-field>
          <v-text-field v-model="filters.abvGt" label="ABV (greater than)" hide-details></v-text-field>
          <v-text-field v-model="filters.ibuLt" label="IBU (less than)" hide-details></v-text-field>
          <v-text-field v-model="filters.ibuGt" label="IBU (greater than)" hide-details></v-text-field>
          <v-text-field v-model="filters.ebcLt" label="EBC (less than)" hide-details></v-text-field>
          <v-text-field v-model="filters.ebcGt" label="EBC (greater than)" hide-details></v-text-field>
        </v-flex>
        <v-flex lg3 md3 sm5>
          <v-text-field v-model="filters.beerName" label="Beer name" hide-details></v-text-field>
          <v-text-field v-model="filters.yeastName" label="Yeast" hide-details></v-text-field>
          <v-text-field v-model="filters.hopsName" label="Hops" hide-details></v-text-field>
          <v-text-field v-model="filters.maltName" label="Malt" hide-details></v-text-field>
          <v-text-field v-model="filters.foodName" label="Food" hide-details></v-text-field>
          <v-text-field v-model="filters.ids" label="IDs" hide-details></v-text-field>
        </v-flex>
        <v-flex lg3 md3 sm12>
          <v-menu
            ref="beforeDatepicker"
            :close-on-content-click="false"
            v-model="beforeDatepicker"
            :nudge-right="40"
            :return-value.sync="filters.beforeDate"
            lazy
            transition="scale-transition"
            offset-y
            full-width
            min-width="250px"
          >
            <v-text-field
              slot="activator"
              v-model="filters.beforeDate"
              label="Brewed before"
              prepend-icon="event"
              readonly
              clearable
              hide-details
            ></v-text-field>
            <v-date-picker
              v-model="filters.beforeDate"
              @input="$refs.beforeDatepicker.save(filters.beforeDate)"
              type="month"
            ></v-date-picker>
          </v-menu>
          <v-menu
            ref="afterDatepicker"
            :close-on-content-click="false"
            v-model="afterDatepicker"
            :nudge-right="40"
            :return-value.sync="filters.afterDate"
            lazy
            transition="scale-transition"
            offset-y
            full-width
            min-width="250px"
          >
            <v-text-field
              slot="activator"
              v-model="filters.afterDate"
              label="Brewed after"
              prepend-icon="event"
              readonly
              clearable
              hide-details
            ></v-text-field>
            <v-date-picker
              v-model="filters.afterDate"
              @input="$refs.afterDatepicker.save(filters.afterDate)"
              type="month"
            ></v-date-picker>
          </v-menu>
        </v-flex>
      </v-layout>
    </v-card-text>
  </v-card>
</template>

<script>
/**
 * Component containing the search filters
 */
export default {
  name: "SearchFilters",
  props: {
    /**
     * Flag to check if filters have been applied
     */
    applyFilters: {
      type: Boolean
    },
    /**
     * Flag to check if filters need to be cleared
     */
    clearFilters: {
      type: Boolean
    }
  },
  watch: {
    applyFilters(val) {
      if (val) {
        /**
         * Triggered when the search button is clicked
         * @event submit:applySearchFilters
         * @type {Object}
         */
        this.$emit("submit:applySearchFilters", { ...this.filters });
      }
    },
    clearFilters(val) {
      if (val) {
        this.filters = {
          abvLt: null,
          abvGt: null,
          ibuLt: null,
          ibuGt: null,
          ebcLt: null,
          ebcGt: null,
          beerName: "",
          yeastName: "",
          beforeDate: null,
          afterDate: null,
          hopsName: "",
          maltName: "",
          foodName: "",
          ids: ""
        };
        this.beforeDatepicker = null;
        this.afterDatepicker = null;
      }
    }
  },
  data() {
    return {
      filters: {
        abvLt: null,
        abvGt: null,
        ibuLt: null,
        ibuGt: null,
        ebcLt: null,
        ebcGt: null,
        beerName: "",
        yeastName: "",
        beforeDate: null,
        afterDate: null,
        hopsName: "",
        maltName: "",
        foodName: "",
        ids: ""
      },
      beforeDatepicker: null,
      afterDatepicker: null
    };
  }
};
</script>

<style lang="scss" scoped>
</style>

<docs>
```jsx
<SearchFilters :style="{height: '200px', overflowY: 'scroll'}" />
```
</docs>