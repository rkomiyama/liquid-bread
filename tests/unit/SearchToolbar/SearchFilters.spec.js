import Vue from "vue";
import Vuetify from "vuetify";
import { shallowMount } from "@vue/test-utils";
import { createRenderer } from "vue-server-renderer";
import SearchFilters from "@/components/SearchToolbar/SearchFilters.vue";

Vue.use(Vuetify);

let wrapper;

describe("SearchFilters.vue", () => {
  beforeEach(() => {
    wrapper = shallowMount(SearchFilters, {
      propsData: {
        applyFilters: false
      }
    });
  });

  it("emits submit:applySearchFilters", () => {
    expect(wrapper.emitted("submit:applySearchFilters")).toBeFalsy();
    wrapper.setProps({ applyFilters: true });
    expect(wrapper.emitted("submit:applySearchFilters")).toBeTruthy();
    wrapper.setProps({ applyFilters: false });
    expect(wrapper.vm.applyFilters).toBeFalsy();
  });

  it("SearchFilters has same HTML structure", () => {
    const renderer = createRenderer();
    renderer.renderToString(wrapper.vm, (err, str) => {
      if (err) throw new Error(err);
      expect(str).toMatchSnapshot();
    });
  });
});
