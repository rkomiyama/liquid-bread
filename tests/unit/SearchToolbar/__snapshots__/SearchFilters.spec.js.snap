// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchFilters.vue SearchFilters has same HTML structure 1`] = `
<v-card-stub tag="div">
  <v-card-title-stub>Search filters</v-card-title-stub>
  <v-card-text-stub class="pt-0">
    <v-layout-stub tag="div">
      <v-flex-stub tag="div" xs1="">
        <v-text-field-stub hide-details=""></v-text-field-stub>
      </v-flex-stub>
      <v-subheader-stub>&lt; ABV &gt;</v-subheader-stub>
      <v-flex-stub tag="div" xs1="">
        <v-text-field-stub hide-details=""></v-text-field-stub>
      </v-flex-stub>
    </v-layout-stub>
    <v-layout-stub tag="div">
      <v-flex-stub tag="div" xs1="">
        <v-text-field-stub hide-details=""></v-text-field-stub>
      </v-flex-stub>
      <v-subheader-stub>&lt; IBU &gt;</v-subheader-stub>
      <v-flex-stub tag="div" xs1="">
        <v-text-field-stub hide-details=""></v-text-field-stub>
      </v-flex-stub>
    </v-layout-stub>
    <v-layout-stub tag="div">
      <v-flex-stub tag="div" xs1="">
        <v-text-field-stub hide-details=""></v-text-field-stub>
      </v-flex-stub>
      <v-subheader-stub>&lt; EBC &gt;</v-subheader-stub>
      <v-flex-stub tag="div" xs1="">
        <v-text-field-stub hide-details=""></v-text-field-stub>
      </v-flex-stub>
    </v-layout-stub>
    <v-text-field-stub label="Beer name" hide-details="" value=""></v-text-field-stub>
    <v-text-field-stub label="Yeast" hide-details="" value=""></v-text-field-stub>
    <v-layout-stub tag="div">
      <v-flex-stub tag="div" xs12="" sm6="" md4="">
        <v-menu-stub openDelay="0" closeDelay="0" lazy="true" contentClass="" maxWidth="auto" minWidth="250px" nudgeBottom="0" nudgeLeft="0" nudgeRight="40" nudgeTop="0" nudgeWidth="0" closeOnClick="true" fullWidth="true" maxHeight="auto" openOnClick="true" offsetY="true" origin="top left" transition="scale-transition">
          <v-text-field-stub label="Brewed before" prepend-icon="event" readonly="readonly" hide-details=""></v-text-field-stub>
          <v-date-picker-stub width="290" eventColor="warning" firstDayOfWeek="0" locale="en-us" nextIcon="$vuetify.icons.next" prevIcon="$vuetify.icons.prev" showCurrent="true" type="month"></v-date-picker-stub>
        </v-menu-stub>
      </v-flex-stub>
      <v-flex-stub tag="div" xs12="" sm6="" md4="">
        <v-menu-stub openDelay="0" closeDelay="0" lazy="true" contentClass="" maxWidth="auto" minWidth="250px" nudgeBottom="0" nudgeLeft="0" nudgeRight="40" nudgeTop="0" nudgeWidth="0" closeOnClick="true" fullWidth="true" maxHeight="auto" openOnClick="true" offsetY="true" origin="top left" transition="scale-transition">
          <v-text-field-stub label="Brewed after" prepend-icon="event" readonly="readonly" hide-details=""></v-text-field-stub>
          <v-date-picker-stub width="290" eventColor="warning" firstDayOfWeek="0" locale="en-us" nextIcon="$vuetify.icons.next" prevIcon="$vuetify.icons.prev" showCurrent="true" type="month"></v-date-picker-stub>
        </v-menu-stub>
      </v-flex-stub>
    </v-layout-stub>
    <v-text-field-stub label="Hops" hide-details="" value=""></v-text-field-stub>
    <v-text-field-stub label="Malt" hide-details="" value=""></v-text-field-stub>
    <v-text-field-stub label="Food" hide-details="" value=""></v-text-field-stub>
    <v-text-field-stub label="IDs" hide-details="" value=""></v-text-field-stub>
  </v-card-text-stub>
</v-card-stub>
`;
